kf/nfconfig.go

```diff
@@ -942,3 +943,311 @@ func getHostInterfaces() (map[string]bool, error) {
 	}
 	return hostIfaces, nil
 }
+
+func (c *NFConfigs) AddAndStartBPF(bpfProg *models.BPFProgram, ifaceName string, direction string) error {
+	var bpfList *list.List
+	if bpfProg == nil {
+		return fmt.Errorf("AddAndStartBPF - bpf program is nil")
+	}
+
+	if bpfProg.AdminStatus == models.Disabled {
+		return nil
+	}
+
+	switch direction {
```

Comment:
Nit: Probably turn this into a 2D lookup table of funcPtr[<direction>][<intfname>] so that you can remove the switch-case.

Reply:
I think it is not necessery because the number of directions are small.

Reply:
Since we want l3afd to be able to work with arbitrary prog types in the future, I think it will be necessary in the future, and "direction" will be a poor label at that point.

Reply:
I agree on the name change, would prefer it to be separate PR as this PR is already very large.  I am thinking something like "attachmentType" would make more sense than direction and should support other use-cases such as kprobes, uprobes, etc..

Reply:
Ok to do in separate PR